packages/backend-core/tests/core/utilities/utils/queue.ts

```diff
@@ -0,0 +1,31 @@
+import { Queue } from "bull"
+import { GenericContainer, Wait } from "testcontainers"
+
+export async function useRealQueues() {
+  const redis = await new GenericContainer("redis")
+    .withExposedPorts(6379)
+    .withEnvironment({})
+    .withLabels({ "com.budibase": "true" })
+    .withReuse()
+    .withWaitStrategy(
+      Wait.forSuccessfulCommand(
+        `until redis-cli  ping | grep -q PONG; do
+          echo "Waiting for Redis to be ready..."
+          sleep 1
+        done
+        echo "Redis is ready!"`
+      ).withStartupTimeout(10000)
+    )
+    .start()
+
+  const port = redis.getMappedPort(6379)
```

Comment:
`getMappedPort` causes problems with dual-stack (v4/v6) systems, see comment [here](https://github.com/Budibase/budibase/blob/ee20f2eba8d8c4a2dcd21661871183787b1d8601/packages/server/src/integrations/tests/utils/index.ts#L220-L228).

I _strongly_ recommend using the `startContainer` function in the linked file. It works around several problems that used to plague us when using `testcontainers`.

Reply:
It this not happening because we are creating the test container and retrieving it separately? Will this happen with using the test container directly?

Reply:
Done!